storage3/_async/file_api.py

```diff
@@ -62,18 +56,67 @@ async def create_signed_url(
             file path to be downloaded, including the current file name.
         expires_in
             number of seconds until the signed URL expires.
+        options
+            options to be passed for downloading or transforming the file.
         """
+        headers = {}
+        token = self._client.headers.get("Authorization")
+        if token:
+            headers["Authorization"] = token
```

Comment:
Correct me if I'm wrong but I don't think this fixes the issue described in #73 - in fact this does nothing, because any headers set in `self._client.headers` gets automatically sent with every request so just taking the token out of there and then passing it explicitly as the `headers` parameter to `.request` does nothing different

The issue described in #73 can be fixed only if we somehow update `self._client.headers` with the new API key generated when a user signs in - and this can only be done upstream. The real solution would be listening for auth state change in the [supabase-py](https://github.com/supabase-community/supabase-py) lib and then updating the `headers` of every "child" client (PostgrestClient, StorageClient etc)

Reply:
Make sense to me now, thanks for this feedback and clarification. I see the fix that's needed. Basically we need to implement the `_listenForAuthEvents()` and `_handleTokenChanged()` functions from supabase-js. I'll handle it in a separate PR.